### PR TITLE
maint(core, query): Upgrade to Monix 3.4.0; Monix-kafka 1.0.0-RC6

### PR DESCRIPTION
**Pull request by vishramachandran**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

Monix library version upgrade. Here are notable changes:

* runAsync rename to runToFuture used in many places
* Many other method renames, some to include "unsafe" suffix to indicate that dual consumption may be an issue.
* monix kafka custom changes not needed anymore. Latest version works for us
* mapAsync became mapParallelUnordered, and we have mapParallelOrdered
* Task.apply became synchronous in this release. These invocations needed to be changed to Task.evalAsync

Before 
```
[info] QueryInMemoryBenchmark.singleThreadedSumRateCCQuery       thrpt   15  2423.331 ± 58.867  ops/s
[info] QueryAndIngestBenchmark.parallelQueries       thrpt   15  2757.459 ± 1068.759  ops/s
```

After
```
[info] QueryInMemoryBenchmark.singleThreadedSumRateCCQuery       thrpt   15  2217.499 ± 247.326  ops/s
[info] QueryAndIngestBenchmark.parallelQueries       thrpt   15  2584.266 ± 1092.581  ops/s
```

Also confirmed that query scheduler did not see increase of number of tasks submitted before and after the monix upgrade.

### CODE REVIEW

**Review by alextheimer — APPROVED** [review]
Thank you for the heads-up.

**Review by amolnayak311 — CHANGES_REQUESTED** [review]
Some comments left, also tests fail.

File: coordinator/src/main/scala/filodb.coordinator/QueryActor.scala [hunk 1]
```diff
@@ -116,7 +116,8 @@ final class QueryActor(memStore: MemStore,
       }
     }
     val executor = new ForkJoinPool( numSchedThreads, threadFactory, exceptionHandler, true)
-    Scheduler.apply(ExecutorInstrumentation.instrument(executor, schedName))
+    Scheduler.apply(ExecutorInstrumentation.instrument(executor, schedName),
+      ExecutionModel.AlwaysAsyncExecution)
```
[the remark this answers is not in the document]
Comment (vishramachandran) [reply coordinator/src/main/scala/filodb.coordinator/QueryActor.scala]: This is now fixed. Thanks much for the team work on identifying that Task.apply changed from being async to synchronous in the new monix release. With that AlwaysAsync is not needed.

File: core/src/main/scala/filodb.core/memstore/IndexBootstrapper.scala [hunk 2]
```diff
@@ -104,7 +104,7 @@ class IndexBootstrapper(colStore: ColumnStore) {
     val start = System.currentTimeMillis()
     Observable.fromIterable(fromHour to toHour).flatMap { hour =>
       colStore.getPartKeysByUpdateHour(ref, shardNum, hour)
-    }.mapAsync(parallelism) { pk =>
+    }.mapParallelOrdered(parallelism) { pk => // preserve order so updates get applied correctly
```
[the remark this answers is not in the document]
Comment (vishramachandran) [reply core/src/main/scala/filodb.core/memstore/IndexBootstrapper.scala]: In theory, downsample part key index update stream can contain dupes, so order is important.
However I realize, dupes are far apart, so Unordered should suffice here. Will change

**Review by amolnayak311 — APPROVED** [review]
LGTM

**Pull request merged**